pkg/lib/plot.js

```diff
@@ -479,19 +481,23 @@ export class Plot {
         this.cur_hover_series = null;
         this.cur_hover_val = false;
 
-        $(this.element).on('plothover', null, this, this.hover_on);
-        $(this.element).on('mouseleave', null, this, this.hover_off);
-        $(this.element).on('plotselecting', null, this, this.selecting);
-        $(this.element).on('plotselected', null, this, this.selected);
+        if (this.element) {
+            $(this.element).on('plothover', null, this, this.hover_on);
+            $(this.element).on('mouseleave', null, this, this.hover_off);
+            $(this.element).on('plotselecting', null, this, this.selecting);
+            $(this.element).on('plotselected', null, this, this.selected);
 
-        // for testing
-        $(this.element).data('flot_data', this.flot_data);
+            // for testing
+            $(this.element).data('flot_data', this.flot_data);
```

Comment:
Right now, this looks up `this.element` using `$()` each time. It's better to chain

```js
$(this.element).on('plothover', null, this, this.hover_on)
  .on('plotselecting', null, this, this.selecting)
  .on('mouseleave', null, this, this.hover_off)
  .on('plotselected', null, this, this.selected)
  .on('plotselecting', null, this, this.selecting)
  .on('plotselected', null, this, this.selected); 
```

..._or_ store the results of $(this.element) in a variable and use that.

Reply:
This is old code that we are moving away from.  I wouldn't worry about it.